### PR TITLE
[sqlite] add doc comments in source, update types

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1763

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `SQLite` API ( https://docs.expo.dev/versions/latest/sdk/sqlite) with minor tweaks and additions to the package source code (like `SQLVoidCallback` type replacement with inline `() => void` type).

This PR also includes few tweaks to the `APISectionInterface` component along with interface method support:

Screenshot 2021-08-09 at 15 13 53

During the conversion I have also adjusted few types and added support for alternative, generic object notation (with test).

Unfortunately this package is lacking a lot of descriptions and documentation comments, but still the autogenerated version should be more complete than the current docs.

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Preview

Screenshot 2021-08-09 at 15 14 38

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-sqlite/src/SQLite.ts [hunk 1]
```diff
@@ -80,6 +80,20 @@ function addExecMethod(db: any): WebSQLDatabase {
   return db;
 }
 
+// @needsAudit @docsMissing
+/**
+ * Open a database, creating it if it doesn't exist, and return a `Database` object. On disk, th
+ * database will be created under the app's [documents directory](../filesystem), i.e.
```
Comment (tsapeta) [inline packages/expo-sqlite/src/SQLite.ts:86]:
```suggestion
 * Open a database, creating it if it doesn't exist, and return a `Database` object. On disk,
 * the database will be created under the app's [documents directory](../filesystem), i.e.
```

**Review by tsapeta — APPROVED** [review]

**Review by bbarthec — APPROVED** [review]
Looking good 👍

File: docs/components/plugins/api/APISectionInterfaces.tsx [hunk 2]
```diff
@@ -3,10 +3,18 @@ import React from 'react';
 import { InlineCode } from '~/components/base/code';
```
Comment (bbarthec) [inline docs/components/plugins/api/APISectionInterfaces.tsx:3]:
```suggestion
import ReactMarkdown from 'react-markdown';

import { InlineCode } from '~/components/base/code';
```

**Comment by Simek** [comment]
@bbarthec Hmm, rebuilding the package files did not produce changes in `js.map` file which is currently in conflict, do you have any ide what have happened there?

I'm going to resolve the conflict manually and then I'm going to overwrite and force push the branch, but this is the first time, when problem like this emerges.

**Pull request merged**